feat: Implements syncing of queue

bornav opened the pull request:
Implements a way for the user to sync the queue across multiple devices - #245

This is a partial implementation of the issue, and only implements the queue syncing(no audiobook support)

deluan-tc reviewed (changes requested):
Hey @bornav, first sorry for the long delay in reviewing this. I really appreciate your effort and patience. 

Also for someone without React/JS experience as you said, this is good!

Anyway, I added my comments, and there are some things I'd like to discuss with you about the implementation. Let me know when you are around in Discord, so we can chat about it.

bornav commented on ui/src/actions/player.js:
```diff
@@ -67,6 +67,16 @@ export const playTracks = (data, ids, selectedId) => {
     data: songs,
   }
 }
+//to be implemented on the Player Side
```
i will merge it into the playTracks
as for the timestamp it is used in the ui/src/common/UpdateQueueButton.js

bornav commented on ui/src/common/UpdateQueueButton.js:
```diff
@@ -0,0 +1,147 @@
+}
+
+UpdateQueueButton.defaultProps = {
+  label: 'Get updated Queue',
```
that was the plan but I couldn't figure out how to make it appear

bornav commented on server/subsonic/responses/responses.go:
```diff
@@ -281,6 +281,7 @@ type User struct {
 	JukeboxRole         bool   `xml:"jukeboxRole,attr"            json:"jukeboxRole"`
 	ShareRole           bool   `xml:"shareRole,attr"              json:"shareRole"`
 	VideoConversionRole bool   `xml:"videoConversionRole,attr"    json:"videoConversionRole"`
+	SyncPlaylist        bool   `xml:"syncPlaylist,attr"           json:"syncPlaylist"`
```
correct, this is not part of the subsonic API, but as for if it is strictly necessary, if we want to allow the user to log in from a device that has no cache and we want that new device to respec the sync settings we are required to store it somewhere, i have put it to the subsonic since it is related to it, but since it is not part of the API it can be  moved, but for my testing it didn't affect other platforms

bornav committed:
Added backend implementation for queue syncing - #245
Added database user row sync_playqueue

bornav committed:
Implemented web handeling of queue sync - #245
Added setting in user pannel for toggeling sync

Addied temporary solution for web sync trigger.(on refresh)

bornav committed:
Modified testcases - #245

bornav commented:
@deluan How should we proceed with this, wait for new player?

zorrobert commented:
Hi, is this still in progress? This would be really useful to me
